src/coreclr/jit/codegenxarch.cpp

```diff
@@ -491,12 +491,23 @@ void CodeGen::genSetRegToConst(regNumber targetReg, var_types targetType, GenTre
 
             if (vecCon->IsAllBitsSet())
             {
-                if ((attr != EA_32BYTE) || compiler->compOpportunisticallyDependsOn(InstructionSet_AVX2))
+                if ((attr == EA_16BYTE) ||
+                    ((attr == EA_32BYTE) && compiler->compOpportunisticallyDependsOn(InstructionSet_AVX2)))
                 {
 #if defined(FEATURE_SIMD)
                     emit->emitIns_SIMD_R_R_R(INS_pcmpeqd, attr, targetReg, targetReg, targetReg);
 #else
                     emit->emitIns_R_R(INS_pcmpeqd, attr, targetReg, targetReg);
+#endif // FEATURE_SIMD
+                    break;
+                }
+                else if (attr == EA_64BYTE)
+                {
+                    assert(compiler->compOpportunisticallyDependsOn(InstructionSet_AVX512F));
+#if defined(FEATURE_SIMD)
+                    emit->emitIns_SIMD_R_R_R_I(INS_vpternlogd, attr, targetReg, targetReg, targetReg, (int8_t)0xFF);
+#else
+                    emit->emitIns_R_R_R_I(INS_vpternlogd, attr, targetReg, targetReg, targetReg, (int8_t)0xFF);
```

Comment:
Just the `FEATURE_SIMD` one should be sufficient.

Since this is an EVEX only instruction and EA_64BYTE is FEATURE_SIMD only, we don't have to worry about the fallback path.

Notably the main reason for the `_SIMD_` variant is to handle legacy RMW vs VEX non-RMW semantics. That is, handling:
```asm
; Legacy RMW
movaps xmm0, xmm1
addps xmm0, xmm2

; VEX non-RMW
vaddps xmm0, xmm1, xmm2
```

Since this is an EVEX only instruction, there isn't any RMW vs non-RMW semantic we have to special case either.